zipkin-thrift/src/main/thrift/com/twitter/zipkin/zipkinCore.thrift

```diff
@@ -168,13 +168,17 @@ struct Endpoint {
   /**
    * Classifier of a source or destination in lowercase, such as "zipkin-web".
    *
-   * Conventionally, when the service name isn't known, service_name = "unknown".
-   *
    * This is the primary parameter for trace lookup, so should be intuitive as
    * possible, for example, matching names in service discovery.
    *
+   * Conventionally, when the service name isn't known, service_name = "unknown".
+   * However, it is also permissible to set service_name = "" (empty string).
+   * The difference in the latter usage is that the span will not be queryable
+   * by service name unless more information is added to the span with non-empty
+   * service name, e.g. an additional annotation from the server.
+   *
    * Particularly clients may not have a reliable service name at ingest. One
```

Comment:
good revision.. I agree that when something isn't reliable, empty is better than pasting "unknown". For example, "" means.. wait a bit longer.. this will clear up! vs "unknown" saying "this may be as good as it gets"